### ISSUE TITLE
Support for "esbuild-jest"

### ISSUE THREAD

**Issue by makoru-hikage** [glennsl/bs-jest#90]
This is for projects using React which requires es6 builds. React projects typically uses ES6 builds.

**Comment by glennsl** [comment]
What would be required for this?

**Comment by makoru-hikage** [comment]
Fortunately, you do not need to do any additional code changes for this. I initially thought you would need to.

I just needed to add this in a Jest config file (or "jest" property if you are storing your Jest config in package.json):

```json
{
    "transform": {
      "^.+\\.jsx?$": "esbuild-jest"
    },
    "transformIgnorePatterns": ["<rootDir>/node_modules/(?!(rescript|@glennsl/bs-jest)/)"]
  }
```

This will require "esbuild-jest" of course. When someone uses `rescript build -with-deps` with this property in `bsconfig.json`:
```json
"package-specs": {
    "module": "es6",
    "in-source": true
  }
```
All res* and ml* files are transpiled into js (or mjs) in ES6 flavor. However, the compiled res/ml files in `node_modules` aren't transpiled, so this thing ` "transformIgnorePatterns": ["<rootDir>/node_modules/(?!(rescript|@glennsl/bs-jest)/)"]` is added to the Jest config.

Shall I add something to the docs?

**Comment by makoru-hikage** [comment]
Please take note that I have not done any test files involving React. I use it here: https://github.com/makoru-hikage/sasquare-reasonml

(Please do not mind that misnomer on my personal project.)

**Comment by glennsl** [comment]
Thanks for the explanation!

> Shall I add something to the docs?

Yes. Please do! As a sub-heading to the troubleshooting section, for example, or just referencing this issue if you're feeling lazy.

**Comment by makoru-hikage** [comment]
Thank you! I shall write a sub-heading, submit a PR and close this upon merging.

**Comment by glennsl** [comment]
Closing this as resolved in #91 then. Thanks again!
